Mathlib/Tactic/Positivity/Core.lean

```diff
@@ -342,6 +343,24 @@ private inductive OrderRel : Type
 end Meta.Positivity
 namespace Meta.Positivity
 
+/-- Given an expression `e`, use the core method of the `positivity` tactic to prove it positive,
+or, failing that, nonnegative; return a boolean (signalling whether the strict or non-strict
+inequality was established) together with the proof as an expression. -/
+def bestResult (e : Expr) : MetaM (Bool × Expr) := do
+  let ⟨u, α, _⟩ ← inferTypeQ' e
+  let zα ← synthInstanceQ q(Zero $α)
+  let pα ← synthInstanceQ q(PartialOrder $α)
+  match ← try? (Meta.Positivity.core zα pα e) with
+  | some (.positive pf) => pure (true, pf)
+  | some (.nonnegative pf) => pure (false, pf)
+  | _ => throwError "could not establish the nonnegativity of {e}"
+
+/-- Given an expression `e`, use the core method of the `positivity` tactic to prove it nonnegative.
+-/
+def proveNonneg (e : Expr) : MetaM Expr := do
+  let (strict, pf) ← bestResult e
+  if strict then mkAppM ``le_of_lt #[pf] else pure pf
```

Comment:
Can we reuse [`Mathlib.Meta.Positivity.Strictness.toNonneg`](https://leanprover-community.github.io/mathlib4_docs/Mathlib/Tactic/Positivity/Core.html#Mathlib.Meta.Positivity.Strictness.toNonneg) here?

Reply:
Only for the second of these functions (`proveNonneg`), as far as I can tell, and I think that function would actually get longer if rewritten to use `Strictness.toNonneg`.